test/integration/connect/envoy/helpers.bash

```diff
@@ -117,7 +117,38 @@ function snapshot_envoy_admin {
   local ENVOY_NAME=$2
 
   docker_wget "http://${HOSTPORT}/config_dump" -q -O - > "./workdir/envoy/${ENVOY_NAME}-config_dump.json"
-  docker_wget "http://${HOSTPORT}/clusters" -q -O - > "./workdir/envoy/${ENVOY_NAME}-clusters.out"
+  docker_wget "http://${HOSTPORT}/clusters?format=json" -q -O - > "./workdir/envoy/${ENVOY_NAME}-clusters.json"
+}
+
+function get_healthy_upstream_endpoint_count {
+  local HOSTPORT=$1
+  local CLUSTER_NAME=$2
+  run retry_default curl -s -f "http://${HOSTPORT}/clusters?format=json"
+  [ "$status" -eq 0 ]
+  echo "$output" | jq --raw-output "
+.cluster_statuses[]
+| select(.name|startswith(\"${CLUSTER_NAME}.default.dc1.internal.\"))
+| .host_statuses[].health_status
+| select(.eds_health_status == \"HEALTHY\")
+| length"
+}
+
+function health_upstream_endpoint_count_matches {
```

Comment:
nit: should this be prefixed with `assert`?

Reply:
I plumbed it this way:
```
assert_upstream_has_healthy_endpoints
+---- retries: health_upstream_endpoint_count_matches until condition or times out
      +---- calls: get_healthy_upstream_endpoint_count and tests the condition once
            +---- retries: curl until a value is returned or times out
```

The intent being that you invoke the `assert_*` version in `verify.bats` files with the intent to block test execution until the condition is eventually met or times out. Under the covers it polls for the current state and tests the condition until success. Under _those_ covers the requests retry until a request is returned (waiting for the envoys to be up, but not necessarily OK).

Reply:
Yeah I don't feel strongly, it's just that the convention here is that helpers that return some intermediate result that can be inspected are just functions while functions that have assertions in and either fail or not depending on some passed in expectation are names `assert_*`.

Even if this isn't being called from an actual test body itself it follows the same pattern as other `assert_*` functions which is why I asked if we think it should be named as one for consistency. No big deal though.

Reply:
Ah I see. I'll rename the intermediate function here to `assert_upstream_has_healthy_endpoints_once` to better reflect how it is used.

Reply:
OK that's pushed.